packages/svelte/src/internal/client/dom/blocks/each.js

```diff
@@ -204,342 +213,170 @@ function each(anchor, flags, get_collection, get_key, render_fn, fallback_fn, re
+	var prev = state;
 
-		var controlled_anchor = (flags & EACH_IS_CONTROLLED) !== 0 && b === 0 ? anchor : null;
+	/** @type {import('#client').EachItem[]} */
+	var to_animate = [];
 
-		pause_effects(effects, controlled_anchor, () => {
-			state.items.length = b;
-		});
-	}
-}
+	/** @type {import('#client').EachItem[]} */
+	var matched = [];
 
-/**
- * Reconcile arrays by the equality of the elements in the array. This algorithm
- * is based on Ivi's reconcilation logic:
- * https://github.com/localvoid/ivi/blob/9f1bd0918f487da5b131941228604763c5d8ef56/packages/ivi/src/client/core.ts#L968
- * @template V
- * @param {Array<V>} array
- * @param {import('#client').EachState} state
- * @param {Element | Comment | Text} anchor
- * @param {(anchor: Node, item: import('#client').MaybeSource<V>, index: number | import('#client').Source<number>) => void} render_fn
- * @param {number} flags
- * @param {any[]} keys
- * @returns {void}
- */
-function reconcile_tracked_array(array, state, anchor, render_fn, flags, keys) {
-	var a_items = state.items;
+	/** @type {import('#client').EachItem[]} */
+	var stashed = [];
 
-	var a = a_items.length;
-	var b = array.length;
+	/** @type {V} */
+	var value;
 
-	/** @type {Array<import('#client').EachItem>} */
-	var b_items = Array(b);
+	/** @type {any} */
+	var key;
 
-	var is_animated = (flags & EACH_IS_ANIMATED) !== 0;
-	var should_update = (flags & (EACH_ITEM_REACTIVE | EACH_INDEX_REACTIVE)) !== 0;
-	var is_controlled = (flags & EACH_IS_CONTROLLED) !== 0;
-	var start = 0;
+	/** @type {import('#client').EachItem | undefined} */
 	var item;
 
-	/** @type {import('#client').Effect[]} */
-	var to_destroy = [];
-	/** @type {Array<import('#client').EachItem>} */
-	var to_animate = [];
-
-	// Step 1 — trim common suffix
-	while (a > 0 && b > 0 && a_items[a - 1].k === keys[b - 1]) {
-		item = b_items[--b] = a_items[--a];
-		anchor = get_first_child(item);
-
-		resume_effect(item.e);
+	if (is_animated) {
+		for (let i = 0; i < array.length; i += 1) {
+			value = array[i];
+			key = get_key(value, i);
+			item = items.get(key);
 
-		if (should_update) {
-			update_item(item, array[b], b, flags);
-		}
-		if (is_animated) {
-			item.a?.measure();
-			to_animate.push(item);
+			if (item !== undefined) {
+				item.a?.measure();
+				to_animate.push(item);
+			}
 		}
 	}
 
-	// Step 2 — trim common prefix
-	while (start < a && start < b && a_items[start].k === keys[start]) {
-		item = b_items[start] = a_items[start];
+	for (let i = 0; i < array.length; i += 1) {
+		value = array[i];
+		key = get_key(value, i);
+		item = items.get(key);
+
+		if (item === undefined) {
+			prev = create_item(
+				current ? get_first_child(current) : anchor,
+				prev,
+				prev.next,
+				value,
+				key,
+				i,
+				render_fn,
+				flags
+			);
+
+			items.set(key, prev);
+
+			matched = [];
+			stashed = [];
+
+			current = prev.next;
+			continue;
+		}
 
+		update_item(item, value, i, flags);
```

Comment:
We now always call this, there was a `should_update` flag before that is missing.